docs/RAG_serving_demo.md

```diff
@@ -0,0 +1,155 @@
+- online updates of the documents and reruning the client query
+
+
+## Building the container image of the model server
+
+The public image of Model Server is enabled for python execution but it has a minimal list of python dependencies. Depending on the use cases, a custom dependencies might need to be included in the image. Follow the instructions from https://github.com/openvinotoolkit/model_server/tree/main/demos/python_demos/rag_chatbot
+
+```
+git clone https://github.com/openvinotoolkit/model_server.git
+cd model_server
+# for ubi base image
+make python_image BASE_OS=redhat OVMS_CPP_DOCKER_IMAGE=registry.connect.redhat.com/intel/openvino-model-server OVMS_CPP_IMAGE_TAG=2024.1-gpu 
+# or for ubuntu22 base image
+make python_image BASE_OS=redhat OVMS_CPP_DOCKER_IMAGE=openvino/model_server OVMS_CPP_IMAGE_TAG=2024.1-gpu
+```
+Push the build image to your image registry.
+
+In case of OpenShift, this build operation can be automated using a BuildConfig resource.
+
+```
+spec:
+  nodeSelector: null
+  output:
+    to:
+      kind: ImageStreamTag
+      name: 'ovms:py'
+  resources: {}
+  strategy:
+    type: Docker
+    dockerStrategy:
+      dockerfilePath: Dockerfile.redhat
+      buildArgs:
+        - name: IMAGE_NAME
+          value: 'registry.connect.redhat.com/intel/openvino-model-server@sha256:b8721a65da98bd9b354680ecb2dca8a4d60a1097babe9822ab264f8e951190f7'
+  postCommit: {}
+  source:
+    type: Git
+    git:
+      uri: 'https://github.com/openvinotoolkit/model_server'
+      ref: main
+    contextDir: demos/python_demos
+```
+It will trigger the build based on provided above Dockerfile and the based image. That image will be automatically pushed to the local cluster container registry in the OpenShift.
+
+## Adding the RAG servable configuration to a configmap
+
+The Model Server configuration including servable configuration needs to be attached to deployed pods. In this demo they will be passed via a configmap which can stored arbitrary files. The operator automatically mounts the contnent of a configmap to the serving pod.
+
+```
+pushd $(pwd)
+cd model_server/demos/python_demos/rag_chatbot/servable_stream
+echo "https://gist.githubusercontent.com/ryanloney/42b8ebe29f95ebd4382ee0b2bb50bea2/raw/cfbb679fefb6babec675c7806254a5fff29a5e6b/aipc.txt" > docs.txt
+oc create configmap rag-demo --from-file=config.json=config.json --from-file=graph.pbtxt=graph.pbtxt --from-file=model.py=model.py \
+--from-file=config.py=config.py --from-file=ov_embedding_model.py=ov_embedding_model.py \
+--from-file=ov_llm_model.py=ov_llm_model.py --from-file=docs.txt=docs.txt
+popd
+```
+
+## Adding Hugging Faces token to a secret and the model name
+The RAG demo can work with several models and it can be defined as an environment variable passed to the pod. Some models might require a token to authorize the model pull from Hugging Faces hub. Extra environment variables can be added to the deployment via a config_map or a secret
```

Comment:
```suggestion
The RAG demo can work with several models and it can be defined as an environment variable passed to the pod. Some models might require a token to authorize the model pull from Hugging Face hub. Extra environment variables can be added to the deployment via a config_map or a secret
```